src/server/dflycmd.cc

```diff
@@ -390,10 +390,23 @@ void DflyCmd::ClusterConfig(CmdArgList args, ConnectionContext* cntx) {
     return rb->SendError("Invalid JSON cluster config", kSyntaxErrType);
   }
 
-  if (!sf_->cluster_config()->SetConfig(json.value())) {
+  auto deleted_slot_ids = sf_->cluster_config()->SetConfig(json.value());
+  if (!deleted_slot_ids) {
     return rb->SendError("Invalid cluster configuration.");
   }
 
+  // Delete old slots data.
+  if (!deleted_slot_ids.value().empty()) {
+    auto cb = [&](auto*) {
+      EngineShard* shard = EngineShard::tlocal();
+      if (shard == nullptr)
+        return;
+
+      shard->db_slice().FlushSlots(deleted_slot_ids.value());
```

Comment:
I cant use std::move here for deleted_slot_ids because this cb is called for all shards

Reply:
That's a good observation. Could you add that as a comment maybe?